google/cloud/security/inventory/inventory_loader.py

```diff
@@ -160,26 +161,37 @@ def _build_pipelines(cycle_timestamp, configs, dao):
 
     Returns:
         List of pipelines that will be run.
-    """
 
+    Raises: inventory_errors.LoadDataPipelineError.
+    """
+    pipelines = []
     crm_api_client = crm.CloudResourceManagerClient()
-    admin_api_client = ad.AdminDirectoryClient()
 
     # The order here matters, e.g. groups_pipeline must come before
     # group_members_pipeline.
-    return [
+    pipelines = [
         load_org_iam_policies_pipeline.LoadOrgIamPoliciesPipeline(
             cycle_timestamp, configs, crm_api_client, dao),
         load_projects_pipeline.LoadProjectsPipeline(
             cycle_timestamp, configs, crm_api_client, dao),
         load_projects_iam_policies_pipeline.LoadProjectsIamPoliciesPipeline(
-            cycle_timestamp, configs, crm_api_client, dao),
-        load_groups_pipeline.LoadGroupsPipeline(
-            cycle_timestamp, configs, admin_api_client, dao),
-        load_group_members_pipeline.LoadGroupMembersPipeline(
-            cycle_timestamp, configs, admin_api_client, dao),
+            cycle_timestamp, configs, crm_api_client, dao)
     ]
 
+    if configs.get('inventory_groups'):
+        if util.can_inventory_groups(configs):
+            admin_api_client = ad.AdminDirectoryClient()
+            pipelines.extend([
+                load_groups_pipeline.LoadGroupsPipeline(
+                    cycle_timestamp, configs, admin_api_client, dao),
+                load_group_members_pipeline.LoadGroupMembersPipeline(
+                    cycle_timestamp, configs, admin_api_client, dao)
+            ])
+        else:
+            raise inventory_errors.LoadDataPipelineError(
```

Comment:
Or...do you just want to log it, and not raise an error?  This way, the pipeline can keep running?